gists/snippets/react.md

```diff
@@ -2,22 +2,22 @@
 
 Defines snippets for common `React` components:
 
-- React-redux components (views)
-- Function UI components
-- Form components
+- React-redux components (typically responsible for loading data in a "View")
+- Function components (a composable, reusable block of functionality)
 
 ### Requirements:
 
-- The functional component snippet uses `React.memo`, which was added in React v16.6.0. If you're using an earlier version of React, this optimization can be safely removed.
```

Comment:
This is a premature optimization that is easy to break. We shouldn't be including it by default, but rather reaching for it only under specific circumstances.

My opinions are largely informed by: https://tkdodo.eu/blog/the-uphill-battle-of-memoization and https://overreacted.io/before-you-memo/